packages/pretty-format/src/__tests__/react.test.js

```diff
@@ -13,29 +13,33 @@ const prettyFormat = require('../');
 const ReactTestComponent = require('../plugins/react_test_component');
 const ReactElement = require('../plugins/react_element');
 
-function assertPrintedJSX(actual, expected, opts) {
-  expect(
-    prettyFormat(
-      actual,
-      Object.assign(
-        {
-          plugins: [ReactElement],
-        },
-        opts,
-      ),
+const prettyFormatElementPlugin = (element, options) =>
+  prettyFormat(
+    element,
+    Object.assign(
+      {
+        plugins: [ReactElement],
+      },
+      options,
     ),
-  ).toEqual(expected);
-  expect(
-    prettyFormat(
-      renderer.create(actual).toJSON(),
-      Object.assign(
-        {
-          plugins: [ReactTestComponent, ReactElement],
-        },
-        opts,
-      ),
+  );
+
+const prettyFormatBothPlugins = (object, options) =>
+  prettyFormat(
+    object,
+    Object.assign(
+      {
+        plugins: [ReactTestComponent, ReactElement],
+      },
+      options,
     ),
-  ).toEqual(expected);
+  );
+
+function assertPrintedJSX(val, formatted, options) {
+  expect(prettyFormatElementPlugin(val, options)).toEqual(formatted);
+  expect(
+    prettyFormatBothPlugins(renderer.create(val).toJSON(), options),
+  ).toEqual(formatted);
```

Comment:
One `expect` requires both plugins to return consistent results, which is necessary but not sufficient. I’ll replace `formatted` with `expected` to make it clearer why 2 `expect` are needed:
* Both plugins could return consistent but incorrect results.
* If one plugin is incorrect (for example, `ReactElement` for one falsey child) then a combined `expect` has a 50% chance of implying the wrong result is expected.

Something less than ideal about `assertPrintedJSX` is if a test fails, the message can’t identify which plugin :(

The test of array of elements caused me to factor out helper functions which I renamed and used in several test cases near the end:
* `formatElement`
* `formatTestObject`